### PR TITLE
Feat: Show Ship Category in Shipyard

### PR DESCRIPTION
**Pull request by mOctave**
**Feature:** This PR implements a feature request detailed and discussed on Discord.

## Feature Details
All I did was add a ship's category to its scrolling panel on the shipyard stats (for both owned and unowned ships).

## UI Screenshots

**Before**
Screenshot 2023-10-10 at 8 20 15 AM

**After**
Screenshot 2023-10-10 at 7 47 58 AM

**Before and After (non-scrolling panels)**
Screenshot 2023-10-10 at 7 50 49 AM
Screenshot 2023-10-10 at 8 18 41 AM

## Usage Examples
N/A

## Testing Done
Opened it up, and checked that it worked properly for a few sample ships.

### Automated Tests Added
N/A

## Performance Impact
N/A

### CODE REVIEW

File: source/ShipInfoDisplay.cpp [hunk 1]
```diff
@@ -143,6 +143,13 @@ void ShipInfoDisplay::UpdateAttributes(const Ship &ship, const PlayerInfo &playe
 
 	attributeHeaderLabels.push_back("model:");
 	attributeHeaderValues.push_back(ship.DisplayModelName());
+
+	if(scrollingPanel)
```
Comment (RisingLeaf) [inline source/ShipInfoDisplay.cpp:147]: why only there?
->comment

File: source/Ship.h [hunk 2]
```diff
@@ -157,6 +157,8 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	const std::string &Description() const;
 	// Get the shipyard thumbnail for this ship.
 	const Sprite *Thumbnail() const;
+	// Get the category of this ship.
+	const std::string &Category() const;
```
Comment (RisingLeaf) [inline source/Ship.h:161]: no need for this, you can just call ship.BaseAttributes().Category()

File: source/ShipInfoDisplay.cpp (hunk 1 above)
Comment (quyykk) [reply source/ShipInfoDisplay.cpp:147]: It would exceed the ship info panel height I believe

**Comment by tibetiroka** [comment]
I think if a ship doesn't have a category, it shouldn't show up on these panels, as it would look quite odd.

While the same could also be said about whitespace-only categories, I don't know if filtering them out would be worth the effort. The default/empty category name `""` shouldn't be allowed, though.

**Comment by TomGoodIdea** [comment]
> I think if a ship doesn't have a category, it shouldn't show up on these panels, as it would look quite odd.

maybe display it as something like "category: None"

**Comment by mOctave** [comment]
Would it be a good idea to use the noun (ship, entity, etc.) if there's no category, or is that overthinking it?

File: source/ShipInfoDisplay.cpp [hunk 3]
```diff
@@ -143,6 +143,17 @@ void ShipInfoDisplay::UpdateAttributes(const Ship &ship, const PlayerInfo &playe
 
 	attributeHeaderLabels.push_back("model:");
 	attributeHeaderValues.push_back(ship.DisplayModelName());
+
+	// Only show the ship category on non-scrolling panels with no risk of overflow
+	if(scrollingPanel)
+	{
+		attributeHeaderLabels.push_back("category:");
+		if(ship.BaseAttributes().Category() == "")
```
Comment (tibetiroka) [inline source/ShipInfoDisplay.cpp:151]:
```suggestion
		if(ship.BaseAttributes().Category().empty())
```

**Review by tibetiroka — CHANGES_REQUESTED** [review]
I think displaying "None" is better than the ship model.

**Comment by quyykk** [comment]
Maybe `"None"` -> `"???"`? Seems more consistent with other parts of the game.

File: source/ShipInfoDisplay.cpp [hunk 4]
```diff
@@ -143,6 +143,17 @@ void ShipInfoDisplay::UpdateAttributes(const Ship &ship, const PlayerInfo &playe
...
+			attributeHeaderValues.push_back("None");
+		else
+			attributeHeaderValues.push_back(ship.BaseAttributes().Category());
```
[the remark this answers is not in the document]
Comment (RisingLeaf) [reply source/ShipInfoDisplay.cpp:154]:
```suggestion
		const string &category = ship.BaseAttributes().Category();
		attributeHeaderValues.push_back(category.empty() ? "None" : category);
```

File: source/ShipInfoDisplay.cpp [hunk 5]
```diff
@@ -143,6 +143,17 @@ void ShipInfoDisplay::UpdateAttributes(const Ship &ship, const PlayerInfo &playe
 
 	attributeHeaderLabels.push_back("model:");
 	attributeHeaderValues.push_back(ship.DisplayModelName());
+
+	// Only show the ship category on non-scrolling panels with no risk of overflow
```
Comment (TomGoodIdea) [inline source/ShipInfoDisplay.cpp:147]:
```suggestion
	// Only show the ship category on scrolling panels with no risk of overflow.
```

File: source/ShipInfoDisplay.cpp [hunk 6]
```diff
@@ -143,6 +143,15 @@ void ShipInfoDisplay::UpdateAttributes(const Ship &ship, const PlayerInfo &playe
 
 	attributeHeaderLabels.push_back("model:");
 	attributeHeaderValues.push_back(ship.DisplayModelName());
+
+	// Only show the ship category on scrolling panels with no risk of overflow.
+	if(scrollingPanel)
+	{
+		attributeHeaderLabels.push_back("category:");
+		const string &category = ship.BaseAttributes().Category();
+		attributeHeaderValues.push_back(category.empty() ? "None" : category);
```
Comment (mOctave) [inline source/ShipInfoDisplay.cpp:152]:
```suggestion
		attributeHeaderValues.push_back(category.empty() ? "???" : category);
```

**Review by TomGoodIdea — APPROVED** [review]

**Review by quyykk — APPROVED** [review]

**Review by tibetiroka — APPROVED** [review]

**Pull request merged**